src/index.ts

```diff
@@ -117,16 +121,36 @@ export class Pipeline implements Serializable {
       ? {
           explicitDependencies: true,
           cache: new Map(),
+          mutator: opts.mutator,
```

Comment:
the `newOpts` are only used to do the final serialization of the contained objects, we shouldn't need a mutator here as it doesn't run on `Step`s anymore? Do I misunderstand?

Reply:
yeah you're right, this isn't needed. will remove

Reply:
removed